rpc/src/rpc.rs

```diff
@@ -3804,6 +3811,24 @@ pub mod rpc_full {
                 None
             };
 
+            let inner_instructions = inner_instructions.map(|info| {
+                info.into_iter()
+                    .enumerate()
+                    .map(|(index, instructions)| InnerInstructions {
+                        index: index as u8,
+                        instructions: instructions
+                            .into_iter()
+                            .map(|info| InnerInstruction {
+                                stack_height: Some(u32::from(info.stack_height)),
+                                instruction: info.instruction,
+                            })
+                            .collect(),
+                    })
+                    .filter(|i| !i.instructions.is_empty())
+                    .map(|converted| UiInnerInstructions::parse(converted, &account_keys))
+                    .collect()
```

Comment:
Since we do this identically (except for the last `map()`) 2 other places already, let's please dedupe them all. The method should probably return `impl Iterator` to do that extra map efficiently, and maybe take an Iterator as input as well.